hollow/src/test/java/com/netflix/hollow/api/consumer/LocalBlobStoreTest.java

```diff
@@ -1,14 +1,131 @@
+        final File localDir = createLocalDir();
+
+        // produce a write state
+        InMemoryBlobStore bs = new InMemoryBlobStore();
+        HollowProducer producer = HollowProducer.withPublisher(bs)
+                .withBlobStager(new HollowInMemoryBlobStager())
+                .build();
+        final long testVersion = producer.runCycle(ws -> {
+            ws.add(1); ws.add(2); ws.add(3);
+        });
+
+        // write state to disk
+        final Path testBlob = Paths.get(localDir.getAbsolutePath(), testVersion + ".blob");
+        HollowBlobWriter blobWriter = new HollowBlobWriter(producer.getWriteEngine());
+        try (OutputStream os = Files.newOutputStream(testBlob)) {
+            blobWriter.writeSnapshot(os);
+        } catch (Exception ex) {
+            throw new IOException("Failed to write snapshot to disk: ", ex);
+        }
+
+        // The first consumer refresh returns mockInputStream (which fails read midway of reading the blob file), successive refreshes return the blob file
+        HollowConsumer.Blob blob = new HollowConsumer.Blob(HollowConstants.VERSION_NONE, testVersion) {
+            @Override public InputStream getInputStream() throws IOException {
+                if (callCount == 0) {
+                    callCount ++;
+                    return mockInputStream;
+                }
+                else {
+                    return Files.newInputStream(testBlob);
+                }
+            }
+        };
+        when(mockBlobRetriever.retrieveSnapshotBlob(anyLong())).thenReturn(blob);
+
+        // The first blob read returns the first byte of blob, the second read throws an exception to simulate failure
+        when(mockInputStream.read(anyObject())).thenAnswer(new Answer() {
+            private int count = 0;
+
+            public Object answer(InvocationOnMock invocation) {
+                if (count++ == 0) { // first invocation
+                    Object[] args = invocation.getArguments();
+                    try {
+                        InputStream is = Files.newInputStream(testBlob);
+                        is.read(((byte[])args[0]), 0, 1);   // read 1 byte
+                        return 1;
+                    } catch (IOException e) {
+                        throw new RuntimeException("Failed to read snapshot file");
+                    }
+                }
+
+                // successive invocations
+                throw new RuntimeException("Simulating download failure");
+            }
+        });
+
+        // Consumer tries to refresh to published version, fails midway
+        HollowConsumer.Builder consumerBuilder = HollowConsumer
+                .withBlobRetriever(mockBlobRetriever)
+                .withLocalBlobStore(localDir);
+        consumerBuilder.noFallBackForExistingSnapshot = false;
+        HollowConsumer consumer = consumerBuilder.build();
+        boolean downloadInterrupted = false;
+        try {
+            consumer.triggerRefreshTo(testVersion);
```

Comment:
minor cleanup -  you could move the `fail` call here and not need the `downloadInterrupted` boolean at all